src/engine/Math/Random.ts

```diff
@@ -48,14 +50,22 @@ export class Random {
   private _mt: number[];
 
   private _index: number;
+  private _seed: number;
 
   /**
    * If no seed is specified, the Date.now() is used
    */
-  constructor(public seed?: number) {
+  constructor(seed?: number) {
     this._mt = new Array<number>(this._n);
     // need to mask to support higher bit machines
-    this._mt[0] = (seed || Date.now()) >>> 0;
+
+    const dynamicSeed = () => {
+      seedOffset++;
+      return initialSeed + seedOffset;
+    };
+
+    this._mt[0] = (seed || dynamicSeed()) >>> 0;
```

Comment:
this is a nitpick but creating an anonymous function here could be detrimental to performance if we're in a hot path. Would rather do `(seed || (initialSeed + seedOffset++))`.

Reply:
Excellent, will fix